### ISSUE TITLE
updating BE makes the game not respond

### ISSUE THREAD

**Issue by sk7725** [Anuken/Mindustry#7028]
**Platform**: *The Windows*

**Build**: *22556, 22560*

![image](https://user-images.githubusercontent.com/58885089/174125248-8f24e882-2b98-4dd7-bd3f-ee5d0069d670.png)
> Java(TM) Platform SE Binary is not responding. 
> Closing the program might result in lost data.
> -> Close program
> -> Wait for it to respond

**Issue**: *Mindustry stops responding when you try to download a new BE ingame*

**Steps to reproduce**: *Check for new updates -> download -> watch as your beloved program dies*

**Link(s) to mod(s) used**: *null*

**Save file**: *i love how the warning keeps getting longer*

If you remove the line above without reading it properly and understanding what it means, I will reap your soul. Even if you're playing on someone's server, you can still save the game to a slot.

**(Crash) logs**: *null*

---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [x] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**
- [x] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by BlueWolf3682** [comment]
I reproduced this issue months before on my laptop. My personal fix was compiling the game myself, as it's really easy.

**Comment by hortiSquash** [comment]
i had this issue months ago as well but builds later than erekir fixed it. Also compiling it myself worked

**Comment by BlueWolf3682** [comment]
I noticed the key issue was when the game restarted. For some reason it won't restart properly, causing a softlock.
Since self compiling makes the restart manual, this is not an issue.

**Comment by sk7725** [comment]
> I noticed the key issue was when the game restarted. For some reason it won't restart properly, causing a softlock. Since self compiling makes the restart manual, this is not an issue.

Automatic restart is not functioning as intended. That alone is intended. Yes, the issue may be avoided with manual compiling because it avoids automatic restarts. But that does not mean the issue does not exist.

**Comment by sk7725** [comment]
"Issue - game crashes when an unit is sapwned"
"Since not spawning units makes the game run fine, this is not an issue."

**Comment by Fabizocker456** [comment]
"Issue - game crashes every startup"
"Since not starting the game up does not induce crashes, this is not an issue"

**Comment by BlueWolf3682** [comment]
> 
> Automatic restart is not functioning as intended. Yes, the issue may be avoided with manual compiling because it avoids automatic restarts. But that does not mean the issue does not exist.

~~i didnt intend for it to come off like that, its still a valid overall issue for BE~~

**Comment by Anuken** [comment]
I got my Windows VM set up today... and it doesn't happen there. I tried a both an outdated build from the official builds, and a custom-built """BE""" built set to update to the latest version, and neither one of these froze the game on Windows.

There was a small bug with the update checker that ignored errors and made it display the load screen forever, but that doesn't appear to be what's occurring here. Either way, I can't reproduce it anywhere, and it's easy enough to just go and download a new build yourself, so... this is not being fixed.
